### ISSUE TITLE
confusing tilt output when attaching to an existing pod

### ISSUE THREAD

**Issue by landism** [tilt-dev/tilt#4566]
### Repro

1. `tilt up` a k8s resource that creates a pod
2. ctrl+c tilt
3. `tilt up` that resource again

### observed
e.g.:
```
Tracking new pod rollout (matt-fortune-6b8dc4dc74-24llm):
     ┊ Scheduled       - <1s
     ┊ Initialized     - <1s
     ┊ Ready           - 2s
```
And tilt does not display any pod logs that predate the most recent `tilt up`

### expected
1. Since Tilt is attaching to an existing pod rather than causing a new pod to be created, it feels misleading that tilt says "new pod rollout", "scheduled", and "initialized" when none of those things are happening.
2. Since Tilt is representing that a new pod is rolling out, it is confusing that it's not logging runtime logs from that pod.

version: ff62061 (0.20.3-dev)

**Comment by landism** [comment]
This can also be reproduced without restarting Tilt - simply `touch` a file in a k8s resource w/ no live_update.

**Comment by landism** [comment]
I took a stab at this in #4556, thinking it'd be cheap.
1. I'm not sure if podmonitor is the right place to do this - theoretically we want to log this once when a build completes and the conditions are right, and it turned out to check the conditions in podmonitor, both in terms of adding/using data that felt weird in a podmonitor and in terms of just having lots of bugs and feeling like I was balancing a house of cards. On one hand, it'd be nicer to log this where the build finishes so we don't have to try to reconstruct that state. On the other hand, this message is intended to be the complement of the existing podmonitor output, so it'd be nice to have here.
2. My attempt at a fix introduced #4594, because it was just checking build finish time and didn't care if it failed. I haven't diagnosed the "logs multiple times" issue, though I've seen that too.
3. I thought a much simpler place to log this message would be at the point that we update [`state.DeployedPodTemplateSpecHashSet`](https://github.com/tilt-dev/tilt/blob/c73534699a5290d52c2163301389138e9ad11441/internal/engine/upper.go#L454), since that's really where the build finishes and we know whether the pod actually changed. IIUC, this works fine for all cases except when the initial build is attaching to an existing pod - at this point we yet don't know what the existing pod's hash spec is, so we can't do anything useful.
